### DISCUSSION TITLE
How can I conditinally set global scss?

### DISCUSSION THREAD

**Discussion by sean256** [vercel/next.js#15240] [Help] [answered]
Problem: I have customer pages in `/pages/customer/*` and some admin stuff in `/pages/admin/*`. Both need a touch of unique global styling applied and I want to use SASS to get all the sassy benefits.

I know I can set global styles using `style jsx` but that doesn't work with external sass files.
```js
import AdminGlobals from './some-sass-file.scss'

export function SomeAdminPage () {
	return (
		<div>
			bla bla bla
			{/* crashes */}
			<style jsx global>{AdminGlobals}</style>
		</div>
	)
}
```

Loading global styles in `_app.js` can't be conditional as far as I can tell.

I can settle on conditionally setting a class on `<body>` using `_document.js`, but I cannot get the router to load so I can see what section of the site is rendering.

Any ideas?

**Comment by amiralies** [comment]
can you explain what do you mean by conditionally?
you can import css / sass like this (for importing sass you need to install `sass` package`)
```js
import './styles.sass'
```

**Reply by sean256** [reply]
If I do that anywhere other than `_app.js` I get this error: `Syntax error: Selector "body" is not pure (pure selectors must contain at least one local class or id)`

I need to be able to load different "global styles" based on what part of the app my users are in.

So if they are `/pages/admin` load admin styles
if they are `/pages/customer` load customer styles

**Comment by jamesmosier** [comment, marked as answer]
This is on the radar for Next, but currently it is not possible the way you outlined. This issue tracks the feature request https://github.com/vercel/next.js/issues/12079. In the meantime, you could look into using CSS modules for component scoped CSS. There is also an RFC for it here https://github.com/vercel/next.js/discussions/15208 

Lastly, I'm not sure what you mean by:
> but I cannot get the router to load so I can see what section of the site is rendering.

Are you just wanting to set a CSS class on the body element based on what page you are on?

**Reply by sean256** [reply]
> Are you just wanting to set a CSS class on the body element based on what page you are on?

That would be an amazing workaround! How can I do that?

**Reply by jamesmosier** [reply]
It really depends on what your requirements are.

You could add it page by page via:

```js
React.useEffect(() => {
  document.getElementsByTagName('body')[0].classList.add('name-of-my-page')
}, [])
```

or if you wanted to do it based on the route, maybe in `_app.js`

```js
import Router from 'next/router'

React.useEffect(() => {
  const handleRouteChange = (url) => {
    if (url === '/admin') {
      document.getElementsByTagName('body')[0].classList.add('admin-page')
    } 

    // ...etc.
  }

  Router.events.on('routeChangeComplete', handleRouteChange)
  return () => {
    Router.events.off('routeChangeComplete', handleRouteChange)
  }
}, [])
```

**Comment by ctsstc** [comment]
I tried something like this for conditionally loading global styling

```javascript
export default function App({ Component, pageProps }) {
  const router = useRouter()
  const newStyle = router.asPath.startsWith('/new')

  if (newStyle) {
    require('../styles/new-custom-bootstrap.scss')
    require('../styles/new-globals.scss')
  } else {
    require('../styles/custom.scss')
    require('../styles/global.scss')
  }

  return (
    <ErrorBoundary>
      <Provider store={store}>
        <Component {...pageProps} />
      </Provider>
    </ErrorBoundary>
  )
}

```

Is is seemingly fine until you navigate from an old page to a new page.

There were a couple of reasons for doing this:
- There was talk about splitting this into two apps at some point later. So I wanted to keep a nice seam between the two parts.
- I wanted to avoid including unused global styling on pages that it was irrelevant to, and to avoid any potential conflicts, or leaking of old styles into the new.

Unfortunately it seems that when you go from an old page such as the login page to the new route it is stuck with the old global styling. If you refresh it will load the new styles and fix the styling issue.

I'm also concerned that if this was ever built out as a static export that this would not work as expected and that ultimately you would want to have two global css files built out and load it for different routes which I imagine is quite outside of what Next JS can handle currently. Maybe some crazy override or handler could be implemented, I'm not sure. I'm not sure if a static export just builds one massive stylesheet that supports the whole site or if each page has its own granular bits. Thus my efficiency concerns may be invalid depending on the implementation of a static site export. 

It's possible my current conditional require would break a static export, or have no styling at all?

I have used the `useEffect` technique with applying a body class to a previous project, so I will likely have to go some route like that with this if I can try to encapsulate the old vs the new.

`#notAnAnswer`

**Reply by lifesmile-ae** [reply]
Hello, I am stuck on this same issue. How did you fix your problem with this ???
